### PR TITLE
[App Service] `az functionapp create`: Update the default image to `mcr.microsoft.com` for Centauri

### PR DESCRIPTION
**Pull request by kamperiadis**
**Related command**
`az functionapp create`

**Description**
For Centauri, if the customer does not provide an image, we provide a default value. This default value needs to be updated.

**Testing Guide**
Create a Centauri function app without an image value and then check that the created function app uses the new default image value.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Review by kaibocai — APPROVED** [review]
LGTM!

**Review by zhoxing-ms — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/appservice/_constants.py [hunk 1]
```diff
@@ -95,4 +95,4 @@ def __init__(self):
     'powershell': 'FunctionApp/windows-powershell-functionapp-on-azure.yml',
 }
 
-DEFAULT_CENTAURI_IMAGE = 'azurefunctionstest.azurecr.io/azure-functions/dotnet7-quickstart-demo:1.0'
+DEFAULT_CENTAURI_IMAGE = 'mcr.microsoft.com/azure-functions/dotnet7-quickstart-demo:1.0'
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/appservice/_constants.py:98]: I'd like to confirm that the position of the default image has changed, but the image itself has not changed, right? It won't cause a breaking change for customers, will it?

Comment (kamperiadis) [reply src/azure-cli/azure/cli/command_modules/appservice/_constants.py:98]: Yes, the image itself has not changed. This default image has only been used by private preview customers for Centauri so even if it is a breaking change, we can communicate that they update the image if they were using the previous default image.

**Comment by zhoxing-ms** [comment]
@kamperiadis Could you please resolve this conflict?

File: src/azure-cli/azure/cli/command_modules/appservice/_constants.py (hunk 1 above)
Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/appservice/_constants.py:98]: Got it, thanks

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**